http.go

```diff
@@ -357,6 +376,7 @@ func NewClient(accountSid string, authToken string, httpClient *http.Client) *Cl
 	c.Queues = &QueueService{client: c}
 	c.Recordings = &RecordingService{client: c}
 	c.Transcriptions = &TranscriptionService{client: c}
+	c.WorkspaceCreator = &WorkspaceCreator{client: c}
```

Comment:
I think there may need to be a little bit of refactoring for this to make architectural sense

Currently this repo has it structured so that things like Workflows, Workers, Queues are all considered to be services initialized by the TaskRouter. While the TaskRouter does interact with them, architecturally those services are associated with a Workspace

So my proposal is for us to move everything that's of the sort `task_router_(service).go` to be `workspace_(service).go` and for there to be a `task_router_workspace.go` which will contain the logic for creating a workspace resource. This will look identical to any of the `task_router_(service).go` file that already exists but this time it will define the `WorkspaceService` struct with the client being the `TaskRouter *client` that has already been defined

I understand it is a bigger refactoring effort and that for our use case whatever is in here can be reviewed and tested separately but if we are to make sense of it all and eventually publish this provider, I think this bit of refactoring would need to happen

Reply:
I think Isaac and I agree, the question should it be for this commit/ PR? Isaac seems to think so, I just don't want to bite off too much in this first increment. I added a test case and fixed up the code itself as well.

Reply:
Not sure if doing the rename/ refactor also will translate well if we try to merge back to the trunk.

Reply:
That's fair. I can review it as it is right now. I don't anticipate us creating any more entries in this repo anyway so prob not worth the trouble